src/main/kotlin/org/opensearch/replication/action/replay/TransportReplayChangesAction.kt

```diff
@@ -166,6 +178,120 @@ class TransportReplayChangesAction @Inject constructor(settings: Settings, trans
         return WriteReplicaResult(request, location, null, replicaShard, log)
     }
 
+    // fetches the index settings from the leader cluster and applies it to the local cluster's index settings.
+    private suspend fun syncRemoteSettings(leaderAlias: String, leaderIndex: String, followerIndex: String) {
```

Comment:
Like discussed offline, let's use common logic between replay and index replication task for mapping and settings sync.

Reply:
ack